src/network/transaction_request/mod.rs

```diff
@@ -18,75 +19,182 @@ pub struct TransactionRequest {
+        self
+    }
+
+    /// Builder-pattern method for building a Zksync EIP-712 create2 tranasaction.
+    pub fn with_create2_params(
+        self,
+        salt: B256,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(Some(salt), code, constructor_data, factory_deps)
+    }
+
+    /// Builder-pattern method for building a ZKsync EIP-712 create tranasaction.
+    pub fn with_create_params(
+        self,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(None, code, constructor_data, factory_deps)
+    }
+
+    pub fn with_create_params_inner(
+        self,
+        salt: Option<B256>,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        let bytecode_hash = hash_bytecode(&code)?;
+        let factory_deps = factory_deps
+            .into_iter()
+            .chain(vec![code])
+            .map(Into::into)
+            .collect();
+        let input = crate::contracts::l2::contract_deployer::encode_create_calldata(
+            salt,
+            bytecode_hash.into(),
+            constructor_data.into(),
+        );
+        Ok(self
+            .with_to(CONTRACT_DEPLOYER_ADDRESS)
+            .with_input(input)
+            .with_factory_deps(factory_deps))
+    }
+}
+
+impl TransactionRequest {
+    #[deprecated(note = "use `set_paymaster_params` instead")]
+    pub fn set_paymaster(&mut self, paymaster_params: PaymasterParams) {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    #[deprecated(note = "use `with_paymaster_params` instead")]
+    pub fn with_paymaster(mut self, paymaster_params: PaymasterParams) -> Self {
+        #[allow(deprecated)]
+        self.set_paymaster(paymaster_params);
+        self
+    }
+
+    #[deprecated(note = "use `with_create_params` instead")]
     pub fn zksync_deploy(
         self,
         code: Vec<u8>,
         constructor_data: Vec<u8>,
         factory_deps: Vec<Vec<u8>>,
     ) -> Result<Self, BytecodeHashError> {
+        #[allow(deprecated)]
         self.zksync_deploy_inner(None, code, constructor_data, factory_deps)
     }
 
+    #[deprecated(note = "use `with_create2_params` instead")]
     pub fn zksync_deploy_with_salt(
         self,
         salt: B256,
         code: Vec<u8>,
         constructor_data: Vec<u8>,
         factory_deps: Vec<Vec<u8>>,
     ) -> Result<Self, BytecodeHashError> {
+        #[allow(deprecated)]
         self.zksync_deploy_inner(Some(salt), code, constructor_data, factory_deps)
     }
 
+    #[deprecated(note = "use `with_create_params_inner` instead")]
     fn zksync_deploy_inner(
```

Comment:
It's a private method -- do we need to have two of them?

Reply:
Ah true, I can update it in place.